kolibri/plugins/video_mp4_render/assets/src/vue/index.vue

```diff
@@ -90,25 +92,34 @@
       },
 
       resizeVideo() {
-        const currentHeight = this.$els.videowrapper.clientHeight;
-        const currentWidth = this.$els.videowrapper.clientWidth;
-        const calcWidth = this.aspectRatio * currentHeight;
-        if (currentWidth < calcWidth) {
-          this.videoPlayer.height(currentWidth / this.aspectRatio);
-          this.videoPlayer.width(currentWidth);
+        const wrapperWrapperWidth = this.$els.videowrapperwrapper.clientWidth;
+        const wrapperWrapperHeight = this.$els.videowrapperwrapper.clientHeight;
+
+        const neededHeightGivenWidth = wrapperWrapperWidth * (1 / this.aspectRatio);
+        const neededWidthGivenHeight = wrapperWrapperHeight * this.aspectRatio;
+
+        let newWidth = 0;
+        let newHeight = 0;
+
+        if (neededHeightGivenWidth <= wrapperWrapperHeight) {
+          newWidth = wrapperWrapperWidth;
+          newHeight = neededHeightGivenWidth;
         } else {
-          this.videoPlayer.width(calcWidth);
-          this.videoPlayer.height(currentHeight);
+          newWidth = neededWidthGivenHeight;
+          newHeight = wrapperWrapperHeight;
         }
+
+        this.$els.videowrapper.setAttribute('style', `width:${newWidth}px;height:${newHeight}px`);
       },
 
       debouncedResizeVideo() {
-        debounce(this.resizeVideo, 300);
+        debounce(this.resizeVideo(), 300);
```

Comment:
This will not debounce, but you are right that the previous implementation was incorrect also.

We need to return the debounced function, rather than just invoking debounce here. Will submit a PR to fix this.